test-data/unit/check-dataclass-transform.test

```diff
@@ -853,7 +853,7 @@ class Person:
     name: str
 
 p = Person('John')
-y = replace(p, name='Bob')  # E: Argument 1 to "replace" has incompatible type "Person"; expected a dataclass
+y = replace(p, name='Bob')
```

Comment:
I think that this can potentially introduce some new weak spots, but on other hand it can eventually work :)

Reply:
This is a false positive but it's due to #15974.